File: frontend/src/types/document.ts
// frontend/src/types/document.ts
/**
 * Represents an Environmental Impact Statement document in the system.
 */
export interface Document {
  /** Unique identifier for the document */
  id: string;
  /** Human-readable name/title of the document */
  title: string;
  /** Optional document description */
  description: string;
  /** Date when the document was created or uploaded */
  date: string;
  /** Optional URL to access the document */
  url: string;
  /** Optional document status */
  status: 'pending' | 'processing' | 'completed' | 'error';
}

/**
 * Response format expected from the backend API when fetching a list of documents.
 */
export interface DocumentsResponse {
  documents: Document[];
}

File: backend/schemas/document.py
# backend/schemas/document.py
from pydantic import BaseModel, Field
from typing import List, Dict, Optional, Any
from datetime import datetime

class Document(BaseModel):
    """
    Represents an Environmental Impact Statement document.
    """
    id: str = Field(..., description="Unique identifier for the document")
    name: str = Field(..., description="Human-readable name of the document")
    date: str = Field(..., description="Document creation or upload date")
    size: int = Field(..., description="Size of the document in bytes")
    url: Optional[str] = Field(None, description="URL to access the document")
    thumbnail_url: Optional[str] = Field(None, description="Thumbnail image URL")
    description: Optional[str] = Field(None, description="Document description")
    type: Optional[str] = Field(None, description="Document type or category")

class DocumentList(BaseModel):
    """
    A list of documents with metadata.
    """
    documents: List[Document] = Field(..., description="List of documents")

class DocumentUpload(BaseModel):
    """
    Response after a document upload.
    """
    id: str = Field(..., description="Assigned document ID")
    name: str = Field(..., description="Original filename")
    size: int = Field(..., description="File size in bytes")
    upload_time: datetime = Field(..., description="Upload timestamp")
    status: str = Field(..., description="Upload status")
